sklego/meta/hierarchical_predictor.py

```diff
@@ -332,11 +350,11 @@ def _predict_estimators(self, X, method_name):
 
     def _fit_single_estimator(self, grp_frame):
         """Shortcut to fit an estimator on a single group"""
-        _X = grp_frame.drop(columns=self.groups_ + [self._TARGET_NAME])
-        _y = grp_frame[self._TARGET_NAME]
+        _X = nw.to_native(grp_frame.drop([*self.groups_, self._TARGET_NAME]))
+        _y = nw.to_native(grp_frame.select(self._TARGET_NAME))
         return clone(self.estimator).fit(_X, _y)
```

Comment:
Here I am ending up with the following issue in case of polars.

If _X is empty (e.g. the dummy case  in `test_expected_output` test), while for pandas dropping all the columns, still maintains the original shape (I guess because index is preserved?), for polars we end up with `_X.shape` being (0, 0).

Which then leads to the estimator raising:

> ValueError: Found input variables with inconsistent numbers of samples

We could say this is kind of an edge case, but users use this as I remember a related issue (and Vincent tutorial on it).

@MarcoGorelli do you have any ideas on how to deal with this?

Reply:
just to have more context, do you happen to remember which issue it was? or the tutorial? I searched "hierarchical empty" here and came out with nothing

numpy does allow for things like `np.zeros((6, 0))`, and I presume things are getting converted to numpy at some point here anyway? So maybe, making an empty numpy array with a given shape somewhere can be a solution?

(I haven't yet looked into this code so sorry if this doesn't make sense)

Reply:
I may also need a reminder 😅 what tutorial are you recalling here?

Reply:
In issue #573 someone reported that specific case